google/data_source_google_active_folder.go

```diff
@@ -35,10 +36,15 @@ func dataSourceGoogleActiveFolderRead(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
-	parent := d.Get("parent").(string)
 	displayName := d.Get("display_name").(string)
 
-	queryString := fmt.Sprintf("lifecycleState=ACTIVE AND parent=%s AND displayName=\"%s\"", parent, displayName)
+	// parent is optional
+	if parent, ok := d.GetOk("parent"); ok {
+		queryString := fmt.Sprintf("lifecycleState=ACTIVE AND parent=%s AND displayName=\"%s\"", parent.(string), displayName)
```

Comment:
You'll need to initialize the variable outside of this if block, and then set a value inside. As-is it isn't in scope when it gets used later.

Reply:
Thank you! This is very helpful, I was wondering whether it was the case.

Reply:
Ah- now, you'll want to change `:=` to `=` on lines 43 and 45. As-is, you're shadowing the `queryString` variable above instead of assigning a value to it.

Reply:
Hey, thank you - fixed that now. This should be good to go I think?